### PR TITLE
chore: add Nx Cloud

### LINKED ISSUE #26700: Enable Nx Cloud for remote caching
We are currently using Nx via Lerna. This gives us the ability to cache the outcome of tasks that we run. This happens locally, but in order for our tasks run in Circle CI to take advantage of caching, we need to use [Nx Cloud](https://nx.dev/nx-cloud/intro/what-is-nx-cloud) so that we can cache our tasks remotely. This will speed up CI run times as we implement task caching with steps like linting, building, testing, etc.

We will be able to use Nx Cloud for free because Cypress is an open source project. We should be able to use [this Nx command](https://nx.dev/packages/nx/documents/connect-to-nx-cloud) to help us generate the configuration for using Nx Cloud.

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes  #26700 

### Additional details
This PR adds [Nx Cloud](https://nx.app/) so that we can share a cache between any targets we define. Nx Cloud also surfaces execution outputs with quick links directly in the terminal and CI as well as in Nx Cloud app itself.  Nx Cloud is free for all Open Source Projects.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by emilyrohrbough — COMMENTED** [review]
How much does this speed up or slow down builds by?

**Comment by jordanpowell88** [comment]
> How much does this speed up or slow down builds by?

This PR won't speed up the execution times at all but it enables us to start enabling that one task at a time

**Review by lmiller1990 — APPROVED** [review]
Looks like a low risk no-op for now, looking forward to better CI runs.

>  Nx Cloud is free for all Open Source Projects.

This is great - hoping it stays this way, if we end up needing to pay, we should consider the value : cost - we pay for a lot of tools, happy to do so, as long as they bring us value. Do we get the full offering for free or are the some pay walled features, even for OSS projects?

**Comment by emilyrohrbough** [comment]
@jordanpowell88 
> This PR won't speed up the execution times at all but it enables us to start enabling that one task at a time?

If this doesn't speed up execution times, what value do we get from enabling this?

**Comment by emilyrohrbough** [comment]
Also if we are sharing cache's what does debugging issues look like?

**Comment by jordanpowell88** [comment]
> This is great - hoping it stays this way, if we end up needing to pay, we should consider the value : cost - we pay for a lot of tools, happy to do so, as long as they bring us value. Do we get the full offering for free or are the some pay walled features, even for OSS projects?

It is a true free offering. They do not intend to change this offering anytime soon either. I don't think we should worry about it but we can always remove it and it will still work locally, it would just slow down CI tasks

**Comment by jordanpowell88** [comment]
> Also if we are sharing cache's what does debugging issues look like?

Outside of the normal output generated by Nx after executing a task, Nx Cloud will generate a link after each task completes that links directly to that run in Nx Cloud.

Example: https://cloud.nx.app/runs/vWUhMNxO0a

**Review by astone123 — APPROVED** [review]

**Comment by jordanpowell88** [comment]
@emilyrohrbough Do you have any more concerns with this change? 

You can see [this PR](https://github.com/cypress-io/cypress/pull/26791) as a reference to how we can slowly start improving CI and local task execution.

File: nx.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "tasksRunnerOptions": {
+    "default": {
+      "runner": "nx-cloud",
+      "options": {
+        "cacheableOperations": [],
+        "accessToken": "NzI2MDMwNzAtZTZmYS00Nzg3LWJhOGMtZWY0ZTg3ZjUwMzQ3fHJlYWQtd3JpdGU="
```
Comment (emilyrohrbough) [inline nx.json]: Is this a read-only token? are we adding a token [override in CI](https://nx.dev/nx-cloud/account/access-tokens#setting-a-different-access-token-in-ci)? When will the cache results be updated? I'd assume on develop runs?

Comment (jordanpowell88) [reply nx.json]: @emilyrohrbough What do you mean by `read-only` token? The results will start immediately after we add each task to the `cacheableOperations` array.

Comment (emilyrohrbough) [reply nx.json]: Right - so if this is token has write access, wont the results from every cypress developer be pushed up and update the cache? per the link I added, it suggested local dev uses read-only and ci uses write-only token.

> It's common to have a read-only token stored in nx.json and a read-write token set via NX_CLOUD_ACCESS_TOKEN in CI.

Comment (jordanpowell88) [reply nx.json]: Ah, yes you are correct. I need to change this

Comment (astone123) [reply nx.json]: Good catch @emilyrohrbough 👀

Comment (jordanpowell88) [reply nx.json]: Updated the local token to be `read-only` in this commit f07c56a786d69caa03c08df95f5ca290095da2e3 and added the `NX_CLOUD_ACCESS_TOKEN` environment variable to be `read-write` in circle

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**